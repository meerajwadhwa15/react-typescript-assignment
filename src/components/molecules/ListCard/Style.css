.card {
  border: 1px solid #000;
  border-radius: 10px;
  color: #2c2c2c;
  line-height: 22px;
  margin-bottom: 20px;
  padding: 15px;
}

.card:hover {
  background: #f2f2f2;
  color: #f2f2f2;
}

.card a {
  display: flex;
  color: #2c2c2c;
  text-decoration: none;
}

.card a > div:first-child {
  flex: 1 1 auto;
}

.card a > div:last-child {
  flex: 3 3 auto;
}
