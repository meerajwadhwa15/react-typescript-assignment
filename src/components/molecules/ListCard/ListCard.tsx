import React, { FunctionComponent } from 'react'
import Img from './../../atoms/Img'
import Anchor from './../../atoms/Anchor'
import './Style.css'

export type Props = {
  item: {
    id: number
    image_url: string
    name: string
    tagline: string
    volume: {
      value: number
      unit: string
    }
  }
}

const BeerList: FunctionComponent<Props> = ({ item }) => (
  <div className="card">
    <Anchor url={`/${item.id}`} title={item.name}>
      <div>
        <Img src={item.image_url} height="250" width="150" alt={item.tagline} />
      </div>
      <div>
        <h2>{item.name}</h2>
        <div>{item.tagline}</div>
        <div>
          {item.volume.value} {item.volume.unit}
        </div>
      </div>
    </Anchor>
  </div>
)

export default BeerList
