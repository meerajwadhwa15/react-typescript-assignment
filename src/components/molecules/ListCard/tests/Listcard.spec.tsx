import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import ListCard, { Props } from '../ListCard'

const props: Props = {
  item: {
    id: 1,
    image_url: 'https://images.punkapi.com/v2/keg.png',
    name: 'Buzz',
    tagline: 'A Real Bitter Experience.',
    volume: {
      value: 20,
      unit: 'litres',
    },
  },
}

describe('ListCard Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(
      <BrowserRouter>
        <ListCard {...props} />
      </BrowserRouter>
    )
  })
})

describe('ListCard Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(
      <BrowserRouter>
        <ListCard {...props} />
      </BrowserRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
