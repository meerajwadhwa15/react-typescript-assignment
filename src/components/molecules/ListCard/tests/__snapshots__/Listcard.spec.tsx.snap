// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard Snapshot testing Matches the snapshot 1`] = `
<div
  className="card"
>
  <a
    href="/1"
    onClick={[Function]}
    title="Buzz"
  >
    <div>
      <img
        alt="A Real Bitter Experience."
        height="250"
        src="https://images.punkapi.com/v2/keg.png"
        width="150"
      />
    </div>
    <div>
      <h2>
        Buzz
      </h2>
      <div>
        A Real Bitter Experience.
      </div>
      <div>
        20
         
        litres
      </div>
    </div>
  </a>
</div>
`;
